File: Android/app/src/main/java/com/swarnkary/donezy/OnboardingScreen.kt
package com.swarnkary.donezy

import androidx.compose.animation.*
import androidx.compose.animation.core.*
import androidx.compose.foundation.ExperimentalFoundationApi
import androidx.compose.foundation.background
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.pager.HorizontalPager
import androidx.compose.foundation.pager.rememberPagerState
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Brush
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import kotlinx.coroutines.launch

@OptIn(ExperimentalFoundationApi::class)
@Composable
fun OnboardingScreen(onComplete: () -> Unit) {
    val pagerState = rememberPagerState(pageCount = { 3 })
    val scope = rememberCoroutineScope()

    val pages = listOf(
        OnboardingPageData(
            title = "Welcome to Donezy",
            description = "Your personal dashboard to track hobbies, form good habits, and structure your daily achievements.",
            emoji = "✨",
            accentBrush = Brush.linearGradient(listOf(Color(0xFF1A6B48), Color(0xFF2D7A46)))
        ),
        OnboardingPageData(
            title = "Smart Reminders & Streaks",
            description = "Stay accountable with customizable reminders, snoozes, and automatic streak tracking for every tracker.",
            emoji = "🔔",
            accentBrush = Brush.linearGradient(listOf(Color(0xFFFF8C00), Color(0xFFFF6347)))
        ),
        OnboardingPageData(
            title = "Unlock Achievements",
            description = "Earn badges, track your insights visually, and share your progress cards directly with friends.",
            emoji = "🏆",
            accentBrush = Brush.linearGradient(listOf(Color(0xFF3F51B5), Color(0xFF00BCD4)))
        )
    )

    Box(
        modifier = Modifier
            .fillMaxSize()
            .background(MaterialTheme.colorScheme.background)
    ) {
        HorizontalPager(
            state = pagerState,
            modifier = Modifier.fillMaxSize()
        ) { page ->
            val data = pages[page]
            Column(
                modifier = Modifier
                    .fillMaxSize()
                    .padding(32.dp),
                horizontalAlignment = Alignment.CenterHorizontally,
                verticalArrangement = Arrangement.Center
            ) {
                // Animated Emoji Float
                val infiniteTransition = rememberInfiniteTransition(label = "emojiFloat")
                val floatAnim by infiniteTransition.animateFloat(
                    initialValue = -8f,
                    targetValue = 8f,
                    animationSpec = infiniteRepeatable(
                        animation = tween(2000, easing = EaseInOutSine),
                        repeatMode = RepeatMode.Reverse
                    ),
                    label = "emojiFloatAnim"
                )

                Box(
                    modifier = Modifier
                        .size(140.dp)
                        .clip(CircleShape)
                        .background(data.accentBrush)
                        .offset(y = floatAnim.dp),
                    contentAlignment = Alignment.Center
                ) {
                    Box(
                        modifier = Modifier
                            .size(136.dp)
                            .clip(CircleShape)
                            .background(MaterialTheme.colorScheme.background)
                            .align(Alignment.Center),
                        contentAlignment = Alignment.Center
                    ) {
                        Text(
                            text = data.emoji,
                            fontSize = 64.sp
                        )
                    }
                }

                Spacer(modifier = Modifier.height(40.dp))

                Text(
                    text = data.title,
                    style = MaterialTheme.typography.headlineMedium,
                    fontWeight = FontWeight.Bold,
                    color = MaterialTheme.colorScheme.onBackground,
                    textAlign = TextAlign.Center
                )

                Spacer(modifier = Modifier.height(16.dp))

                Text(
                    text = data.description,
                    style = MaterialTheme.typography.bodyLarge,
                    color = MaterialTheme.colorScheme.onSurfaceVariant,
                    textAlign = TextAlign.Center,
                    lineHeight = 24.sp
                )
            }
        }

        // Bottom Controls
        Row(
            modifier = Modifier
                .fillMaxWidth()
                .align(Alignment.BottomCenter)
                .padding(32.dp),
            horizontalArrangement = Arrangement.SpaceBetween,
            verticalAlignment = Alignment.CenterVertically
        ) {
            // Page Indicators
            Row(horizontalArrangement = Arrangement.spacedBy(6.dp)) {
                repeat(3) { index ->
                    val active = pagerState.currentPage == index
                    val width by animateDpAsState(
                        targetValue = if (active) 20.dp else 8.dp,
                        animationSpec = spring(dampingRatio = 0.6f, stiffness = 300f),
                        label = "indicatorWidth"
                    )
                    val color = if (active) MaterialTheme.colorScheme.primary else MaterialTheme.colorScheme.outlineVariant
                    Box(
                        modifier = Modifier
                            .height(8.dp)
                            .width(width)
                            .clip(CircleShape)
                            .background(color)
                    )
                }
            }

            // Buttons
            Row(horizontalArrangement = Arrangement.spacedBy(12.dp)) {
                if (pagerState.currentPage < 2) {
                    TextButton(onClick = {
                        scope.launch { pagerState.animateScrollToPage(2) }
                    }) {
                        Text("Skip")
                    }
                    Button(
                        onClick = {
                            scope.launch { pagerState.animateScrollToPage(pagerState.currentPage + 1) }
                        },
                        shape = RoundedCornerShape(12.dp)
                    ) {
                        Text("Next")
                    }
                } else {
                    Button(
                        onClick = onComplete,
                        shape = RoundedCornerShape(12.dp),
                        colors = ButtonDefaults.buttonColors(containerColor = MaterialTheme.colorScheme.primary)
                    ) {
                        Text("Get Started")
                    }
                }
            }
        }
    }
}

data class OnboardingPageData(
    val title: String,
    val description: String,
    val emoji: String,
    val accentBrush: Brush
)

File: MacOs/Donezy/OnboardingScreen.swift
import SwiftUI

struct OnboardingScreen: View {
    let onComplete: () -> Void
    @Environment(\.theme) private var theme
    
    @State private var currentPage = 0
    @State private var floatOffset: CGFloat = -8
    
    private let pages = [
        OnboardingPageData(
            title: "Welcome to Donezy",
            description: "Your personal dashboard to track hobbies, form good habits, and structure your daily achievements.",
            emoji: "✨",
            colors: [Color(hex: 0x1A6B48), Color(hex: 0x2D7A46)]
        ),
        OnboardingPageData(
            title: "Smart Reminders & Streaks",
            description: "Stay accountable with customizable reminders, snoozes, and automatic streak tracking for every tracker.",
            emoji: "🔔",
            colors: [Color(hex: 0xFFFF8C00), Color(hex: 0xFFFF6347)]
        ),
        OnboardingPageData(
            title: "Unlock Achievements",
            description: "Earn badges, track your insights visually, and share your progress cards directly with friends.",
            emoji: "🏆",
            colors: [Color(hex: 0x3F51B5), Color(hex: 0x00BCD4)]
        )
    ]
    
    var body: some View {
        ZStack {
            theme.background.ignoresSafeArea()
            
            VStack(spacing: 0) {
                TabView(selection: $currentPage) {
                    ForEach(0..<pages.count, id: \.self) { index in
                        pageView(for: pages[index])
                            .tag(index)
                    }
                }
                .tabViewStyle(PageTabViewStyle(indexDisplayMode: .never))
                
                // Bottom Controls
                HStack {
                    // Page Indicators
                    HStack(spacing: 6) {
                        ForEach(0..<pages.count, id: \.self) { index in
                            let active = currentPage == index
                            Capsule()
                                .fill(active ? theme.primary : theme.onSurfaceVariant.opacity(0.3))
                                .frame(width: active ? 20 : 8, height: 8)
                                .animation(.spring(response: 0.3, dampingFraction: 0.6, blendDuration: 0), value: currentPage)
                        }
                    }
                    
                    Spacer()
                    
                    // Buttons
                    HStack(spacing: 12) {
                        if currentPage < 2 {
                            Button(action: {
                                withAnimation {
                                    currentPage = 2
                                }
                            }) {
                                Text("Skip")
                                    .font(.system(size: 15, weight: .medium))
                                    .foregroundColor(theme.primary)
                                    .padding(.horizontal, 16)
                                    .padding(.vertical, 8)
                            }
                            .buttonStyle(.plain)
                            
                            Button(action: {
                                withAnimation {
                                    currentPage += 1
                                }
                            }) {
                                Text("Next")
                                    .font(.system(size: 15, weight: .semibold))
                                    .foregroundColor(theme.onPrimary)
                                    .padding(.horizontal, 20)
                                    .padding(.vertical, 10)
                                    .background(theme.primary, in: RoundedRectangle(cornerRadius: 12))
                            }
                            .buttonStyle(.plain)
                        } else {
                            Button(action: onComplete) {
                                Text("Get Started")
                                    .font(.system(size: 15, weight: .bold))
                                    .foregroundColor(theme.onPrimary)
                                    .padding(.horizontal, 24)
                                    .padding(.vertical, 12)
                                    .background(theme.primary, in: RoundedRectangle(cornerRadius: 12))
                            }
                            .buttonStyle(.plain)
                        }
                    }
                }
                .padding(.horizontal, 32)
                .padding(.bottom, 40)
            }
        }
        .onAppear {
            withAnimation(.easeInOut(duration: 2.0).repeatForever(autoreverses: true)) {
                floatOffset = 8
            }
        }
    }
    
    @ViewBuilder
    private func pageView(for data: OnboardingPageData) -> some View {
        VStack(spacing: 0) {
            Spacer()
            
            // Floating emoji container
            ZStack {
                Circle()
                    .fill(LinearGradient(
                        colors: data.colors,
                        startPoint: .topLeading,
                        endPoint: .bottomTrailing
                    ))
                    .frame(width: 140, height: 140)
                
                Circle()
                    .fill(theme.background)
                    .frame(width: 136, height: 136)
                
                Text(data.emoji)
                    .font(.system(size: 64))
            }
            .offset(y: floatOffset)
            
            Spacer().frame(height: 40)
            
            Text(data.title)
                .font(.system(size: 28, weight: .bold))
                .foregroundColor(theme.onBackground)
                .multilineTextAlignment(.center)
                .padding(.horizontal, 32)
            
            Spacer().frame(height: 16)
            
            Text(data.description)
                .font(.system(size: 16))
                .foregroundColor(theme.onSurfaceVariant)
                .multilineTextAlignment(.center)
                .lineSpacing(4)
                .padding(.horizontal, 32)
            
            Spacer()
        }
    }
}

struct OnboardingPageData {
    let title: String
    let description: String
    let emoji: String
    let colors: [Color]
}
